### ISSUE TITLE
Projucer 5 writes incorrect `-include` statement in Linux makefiles

### ISSUE THREAD

**Issue by artfwo** [WeAreROLI/JUCE#202]
Changing header files in JUCE projects on Linux doesn't result in recompiling dependent source files, when the project is built with make.

The issue has been also reported on the forum:
https://forum.juce.com/t/juce-5-linux-recompiling-headers-makefile-encoding/22129

Makefiles produced with Projucer 5 have several targets with different object lists, e.g. `OBJECTS_APP`, `OBJECTS_VST`, `OBJECTS_SHARED_CODE`, etc.

The bottom line of a Makefile looks like:
`-include $(OBJECTS:%.o=%.d)`

Since there's no `OBJECTS` variable to refer to, make fails to include dependencies for the project source files.

**Comment by hogliux** [comment]
Fixed with commit 	[3716bc6](https://github.com/WeAreROLI/JUCE/commit/3716bc6f6637e4aaee85ca4a7b8fcc07e9fc8bd8).
